### ISSUE TITLE
feature request - prettier CLI to resolve aliases and process linked files and pick right extensions, by target file type

### ISSUE THREAD

**Issue by revelt** [prettier/prettier#6165]
**Environments:**
- Prettier Version: 1.17.1
- Running Prettier via: CLI
- Runtime: Node 12.3.1
- Operating System: macOS 10.14.5

**Steps to reproduce:**
Prettier is ran on projects. Projects are sometimes npm packages. npm packages sometimes contain symlinked files, for development purposes. Is it possible to make prettier CLI to resolve the alias files (like https://www.npmjs.com/package/alias-resolver does) and process them correctly? Currently raw source of alias is processed (errors out).

**Desired behavior:**
For example, my application's main entry point, `cli.js` is an alias to same-named real `cli.js`. Is it possible to be able to run `prettier '*.{js,css,scss,vue,md}' --write` and prettier to process the contents of the real `cli.js`? Now prettier's wildcard does pick up the alias but it processes the raw contents of alias file (which is rubbish characters).

Imagine,

`prettier '*.{js,css,scss,vue,md}' --write` catches alias file `cli.js` --> pings its source, real `cli.js` --> processes the real `cli.js`

The tricky part is, aliases can be named any way user wants, so CLI globs have to resolve check what each alias points to, to evaluate those files, not the name of alias file itself (which is system file, full of rubbish characters).

**Comment by duailibe** [comment]
I'm having a hard time understanding what you want. Can you put up a repository already with the links set up and detailed commands that reproduce the behavior you're facing?

**Comment by revelt** [comment]
@duailibe As you know, on Mac, you can right-click and do "make alias" on a file. Try yourself, open your project managed by prettier. Make an alias of one of the file prettier would normally pick up. Name alias `zzz` (no extension). Move original file to the desktop. I'm asking for Prettier to resolve that alias and process the file you moved to desktop.

**Comment by revelt** [comment]
I checked, we're using `globby` and it uses `fast-glob` which in turn seems to be able to resolve aliased files.. It just globby is not resolving aliases..

**Comment by hawkrives** [comment]
@revelt Are you asking for macOS "aliases", or POSIX "symlinks"? Because those are two very, very different beasts.

**Comment by revelt** [comment]
@hawkrives I meant "Symbolic Links", sorry for my ignorance. Lerna monorepos are wired with symbolic links, just checked. I moved the target file and they become broken, which means it's not aliases.

I did some testing, it seems that Mac aliases are not resolving and Prettier will process the system file itself. But if one creates a real symlink (`ln -s` [recipe](https://stackoverflow.com/questions/1951742/how-to-symlink-a-file-in-linux)) then it does resolve, as long as symlink is named with appropriate extension which would be picked up by Prettier.

For example, I moved `README.md` out of one of my projects that Prettier processes. Then created symlink, `ln -s` to it. As long as symlink is called `README.md`, Prettier wildcard will pick it up, resolve it and process the target.

I'm fine with symlinks and with naming them properly so I'll close. Symlinks already work, it was aliases that didn't work.

Thank you. Closing.
